controllers/common/models.go

```diff
@@ -215,3 +215,27 @@ func GetPodDisruptionBudgetModel(f string) (*policyv1.PodDisruptionBudget, error
 	}
 	return rb, nil
 }
+
+func CompileEnvironmentVariables(allEnv []corev1.EnvVar, operatorEnv map[string]string) []corev1.EnvVar {
+	res := []corev1.EnvVar{}
+	for _, env := range allEnv {
+		value, isOperatorEnv := operatorEnv[env.Name]
+		if isOperatorEnv {
+			if env.Value == "" { // set only if no value already set
+				env.Value = value
+			}
+			delete(operatorEnv, env.Name)
```

Comment:
This approach might come in in handy to add missing mandatory env variables, but is does not protect the user against duplicates. Maybe this is not of high importance, but it could prove helpful on the long run.

On the other hand, maybe adding debug logs for env vars where the user override the "default" operator env value could be useful as well. (Maybe log spamming would not be an issue if a low priority log level could be used.) What do you think?

Reply:
right, not sure how to handle duplicates. But I guess this would be a problem that can be solved by the users themselves since it a configuration problem.

I am not sure if I understood the second paragraph, users can set the operator log level which will be reflected to all Meridio pods except if they add it directly to any of the template env variables (since with this PR the non empty env variables will not be overwritten by the operator)

Reply:
Yeah, I just wanted to avoid problems because of duplicates, but since the main source of "truth" will be the template files themself, there's no point wasting time on it.

As for the logging, I meant logging when the template "orders" the operator to use a different value compared to what the operator would have set. Not sure it would help that much when dealing with non-working deployments.

Reply:
ha ok, I understand. Not sure if it would help, I think it is still possible to compare the template and the result when deployed with `kubectl describe pod`/`kubectl get pod`